dlt/destinations/impl/databricks/sql_client.py

```diff
@@ -65,6 +66,18 @@ def rollback_transaction(self) -> None:
     def native_connection(self) -> "DatabricksSqlConnection":
         return self._conn
 
+    def has_dataset(self) -> bool:
+        query = """
+    SELECT CATALOG_NAME
```

Comment:
What's the context on this? Would be great to have a comment describing why this override is necessary. Looks like an odd case that `SELECT 1` doesn't work.  
I also cannot reproduce the false-positive issue with the original `has_dataset` in our databricks. Is this something that happens now with new clusters only?

Reply:
hey @steinitzu thanks for the comments! I've tested it again with the new dependencies, it works without changing has_dataset. the underlying `databricks_sql_connector` had some changes in the way they were parsing the results of the query, causing it to fail. This change was initially an attempt to get the lib to work properly (but only solved the symptom not the root cause)

>   File "databricks\sql\client.py", line 670, in fetchall
    return self.active_result_set.fetchall()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "databricks\sql\client.py", line 944, in fetchall
    return self._convert_arrow_table(self.fetchall_arrow())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "databricks\sql\client.py", line 884, in _convert_arrow_table
    res = df.to_numpy(na_value=None)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pandas\core\frame.py", line 1993, in to_numpy
    result = self._mgr.as_array(dtype=dtype, copy=copy, na_value=na_value)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pandas\core\internals\managers.py", line 1703, in as_array
    arr[isna(arr)] = na_value
    ~~~^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'